tests/lib/tools/snapd.tool

```diff
@@ -47,8 +47,14 @@ main() {
 		esac
 	done
 
-	LIBEXEC_DIR="$(os.paths libexec-dir)"
-	exec "$LIBEXEC_DIR/snapd/$tool" "$@"
+	# Use the tool from the snapd snap when re-exec is enabled
+	# Otherwise use the tool from the distro installation
+	if tests.info is-reexec-done; then
+		exec /snap/snapd/current/usr/lib/snapd/"$tool" "$@"
```

Comment:
I think this will break when we start to re-exec on distros that ues /var/lib/snapd/snap. Can you use the snapd mount dir instead of hard-coded /snap/snapd/current/ please?

Reply:
done